inference.py

```diff
@@ -323,61 +284,77 @@ def main(params: dict):
                                                  fill=0)  # background value in rasterized vector.
+
+                    pred, gdf = segmentation(img_array, raster, label, num_classes_corrected,
+                                             gpkg_name, model, chunk_size, num_bands, device)
+                    if gdf is not None:
+                        gdf_.append(gdf)
+                        gpkg_name_.append(gpkg_name)
+                    if local_gpkg:
                         with start_run(run_name=img_name, nested=True):
                             pixelMetrics= ComputePixelMetrics(label, pred, num_classes_corrected)
-                            log_metrics(pixelMetrics.update(pixelMetrics.jaccard))
+                            log_metrics(pixelMetrics.update(pixelMetrics.iou))
                             log_metrics(pixelMetrics.update(pixelMetrics.dice))
-                            log_metrics(pixelMetrics.update(pixelMetrics.accuracy))
-                            log_metrics(pixelMetrics.update(pixelMetrics.precision))
-                            log_metrics(pixelMetrics.update(pixelMetrics.recall))
-                            log_metrics(pixelMetrics.update(pixelMetrics.matthews))
-
-                        label_classes = np.unique(label)
-                        assert len(colors) >= len(label_classes), f'Not enough colors and class names for number of classes in output'
-                        # FIXME: color mapping scheme is hardcoded for now because of memory constraint; To be fixed.
-                        label_rgb = ind2rgb(label, colors)
-                        pred_rgb = ind2rgb(pred, colors)
-                        Image.fromarray((label_rgb).astype(np.uint8), mode='RGB').save(os.path.join(working_folder, 'label_rgb_' + inference_image.stem + '.png'))
-                        Image.fromarray((pred_rgb).astype(np.uint8), mode='RGB').save(os.path.join(working_folder, 'pred_rgb_' + inference_image.stem + '.png'))
-                        del label_rgb, pred_rgb
-                    pred = pred[np.newaxis, :, :]
+                    pred = pred[np.newaxis, :, :].astype(np.uint8)
                     inf_meta.update({"driver": "GTiff",
                                      "height": pred.shape[1],
                                      "width": pred.shape[2],
                                      "count": pred.shape[0],
                                      "dtype": 'uint8'})
-
                     with rasterio.open(inference_image, 'w+', **inf_meta) as dest:
                         dest.write(pred)
-        log_artifact(working_folder)
+        if len(gdf_) >= 1:
+            assert len(gdf_) == len(gpkg_name_), 'benchmarking unable to complete'
+            all_gdf = pd.concat(gdf_)
+            all_gdf.reset_index(drop=True, inplace=True)
+            gdf_x = gpd.GeoDataFrame(all_gdf)
+            gdf_x.to_file(working_folder.joinpath("benchmark.gpkg"), driver="GPKG", index=False)
```

Comment:
could you add one or two comments on what is going on here. I have a vague idea but since the gdf is populated higher up in the for loop, I think comments would be useful

Reply:
basically, we are concatenating all the benchmarks for each individual image into one larger table. I would add comments!